Runtime/Scripts/EdgeEventsManager.cs

```diff
@@ -453,51 +470,30 @@ IEnumerator RespondToLatencyRequest()
       }
     }
 
-    async Task ProcessLatency(Statistics stats)
+    //Compares between the FindCloudletPerformance Cloudlet and the Current Cloudlet(last stats received from the server)
+    async Task CompareLatencies(FindCloudletReply fcPerformanceReply, Statistics receivedStats)
     {
-      Logger.Log("Processing Latency Received from Server");
-      if (stats.Avg > config.latencyThresholdTriggerMs)
+      if (FCPerformanceReply == null)
       {
-        Logger.Log("Current Latency (" + stats.Avg + ") > LatencyThreshold (" + config.latencyThresholdTriggerMs + ")");
-        Logger.Log("Performing FindCloudlet PerformanceMode");
-        MobiledgeXIntegration tempMxi = new MobiledgeXIntegration();
-        bool regResult = await tempMxi.Register(hostOverride, (uint)portOverride);
-        if (!regResult)
-        {
-          PropagateError(FindCloudletEventTrigger.LatencyTooHigh, "FindCloudletPerformanceMode failed (RegisterClient failed)");
-          return;
-        }
-        tempMxi.UseFindCloudletPerformanceMode(true);
-        bool fcSuccess = await tempMxi.FindCloudlet(hostOverride, (uint)portOverride);
-        if (!fcSuccess)
-        {
-          PropagateError(FindCloudletEventTrigger.LatencyTooHigh, "FindCloudletPerformanceMode failed");
-          return;
-        }
-        currentFindCloudlet = tempMxi.latestFindCloudletReply;
-        tempMxi.Dispose();
-        if (!integration.latestFindCloudletReply.Equals(currentFindCloudlet))
-        {
-          PropagateError(FindCloudletEventTrigger.LatencyTooHigh,
-            "New Cloudlet obtained from FindCloudletPerformanceMode is the same as old cloudlet");
-          return;
-        }
-        else
-        {
-          await CompareLatencies(stats);
-        }
+        PropagateError(FindCloudletEventTrigger.LatencyTooHigh,
+        "Error In FindCloudlet Perfromance Mode, FindCloudlet Reply is null");
+        return;
       }
-      else
+
+      if (FCPerformanceReply.Status == FindCloudletReply.Types.FindStatus.FindNotfound || FCPerformanceReply.Status == FindCloudletReply.Types.FindStatus.FindUnknown)
       {
-        Logger.Log("Current Latency (" + stats.Avg + ") < LatencyThreshold (" + config.latencyThresholdTriggerMs + ")");
+        PropagateError(FindCloudletEventTrigger.LatencyTooHigh,
+       "Error In FindCloudlet Perfromance Mode, FindCloudlet Status is " + FCPerformanceReply.Status);
+        return;
+      }
+
+      if (!integration.latestFindCloudletReply.Equals(FCPerformanceReply))
```

Comment:
I think this may be incorrect. The EdgeEventsCookie will change, so this will always return false. I think all you have to do is compare fqdn